packages/tests/src/email.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  MailinatorClient,
+  GetInboxRequest,
+  GetMessageRequest,
+  Sort,
+  Message,
+  Part,
+} from "mailinator-client";
+import { load } from "cheerio";
+
+export const TEAM_DOMAIN = "team336427.testinator.com";
+
+export type FetchLatestEmailInput = {
+  inbox?: string;
+};
+
+export async function fetchLatestEmail({
+  inbox = "e2e",
+}: FetchLatestEmailInput) {
+  try {
+    const mailinatorClient: MailinatorClient = new MailinatorClient(
+      process.env.MAILINATOR_API_KEY || ""
+    );
+
+    const response = await mailinatorClient.request(
+      new GetInboxRequest(TEAM_DOMAIN, inbox, 0, 1, Sort.DESC, true)
+    );
+
+    if (
+      !response.result ||
+      !response.result.msgs ||
+      response.result.msgs.length === 0
+    ) {
+      return null;
+    }
+
+    return response.result.msgs[0];
+  } catch (e) {
+    console.log("Error fetching email", e);
+    return null;
+  }
+}
+
+export type GetMessageInput = {
+  messageId: string;
+};
+
+export async function getMessage({ messageId }: GetMessageInput) {
+  try {
+    const mailinatorClient: MailinatorClient = new MailinatorClient(
+      "4c39b2cfe3c746a09926ca6e0a17c779"
+    );
+
+    const response = await mailinatorClient.request(
+      new GetMessageRequest(TEAM_DOMAIN, messageId)
+    );
+
+    if (!response.result) {
+      return null;
+    }
+
+    return response.result;
+  } catch (e) {
+    console.log("Error fetching email", e);
+    return null;
+  }
+}
+
+export function getTextContent(message: Message) {
+  const messageContent = message.parts.reduce((acc: string, part: Part) => {
+    if (part.body) {
+      return acc + part.body;
+    }
+    return acc;
+  }, "");
+
+  return messageContent;
+}
+
+export function getTheJumpPageURL(email: string) {
+  const $ = load(email);
+  const jumpPageLink = $("a:contains('Confirm')").attr("href");
+  return jumpPageLink;
+}
+
+export async function fetchLatestEmailWithRetry(
+  input: FetchLatestEmailInput
+): Promise<Message | null> {
+  const maxRetries = 5;
```

Comment:
Emails are slow and asynchronous. 5 retries doesn't seem to be enough.

My guess is that it should retry for at least 30s, possibly more.

Reply:
Agreed, I did not tweak these things - just wanted to get a running baseline for now.